velox/vector/BaseVector.h

```diff
@@ -863,6 +863,19 @@ class BaseVector {
     storageByteCount_ = std::nullopt;
   }
 
+  /// Slice a buffer with specific type.
+  /// For boolean type and if the 'offset' is not multiple of 8, return a
+  /// shifted copy, new buffer is allocated from 'pool'.
+  /// Otherwise return a BufferView into the original buffer (with shared
+  /// ownership of original buffer).
+  /// @param type Must be primitive type or OPAQUE type.
```

Comment:
Looks like it is sufficient to pass TypeKind, Type is not needed.

Would you document all parameters?

Reply:
It's using a few utilities provided in `Type` class, would need to duplicate the logic if we have only `TypeKind`.  I would say we can keep it for now until a use case showing up requiring `TypeKind` only (the possibility would be low).

Reply:
> Would you document all parameters?

Done, thanks.